eval/topdown.go

```diff
@@ -0,0 +1,786 @@
+// Copyright 2016 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package eval
+
+import "github.com/open-policy-agent/opa/opalog"
+import "fmt"
+
+var _ = fmt.Println
+
+// Bindings is the structure used to ground expressions.
+type Bindings map[opalog.Var]opalog.Value
```

Comment:
Unfortunately this will change in the next PR as the implementation binds refs to values when evaluating virtual docs.